dev.py

```diff
@@ -16,63 +16,94 @@
+    raise SystemError("No support for Python 3.x under 3.5.")
+
+
+import os, shutil, sys, subprocess, tempfile
 
 for d in 'eggs', 'develop-eggs', 'bin', 'parts':
     if not os.path.exists(d):
         os.mkdir(d)
 
+bin_buildout = os.path.join('bin', 'buildout')
+if os.path.isfile(bin_buildout):
+    os.remove(bin_buildout)
+
 if os.path.isdir('build'):
     shutil.rmtree('build')
 
-######################################################################
-# Make sure we have a relatively clean environment
-try:
-    import pkg_resources, setuptools
-except ImportError:
-    pass
-else:
-    raise SystemError(
-        "Buildout development with a pre-installed setuptools or "
-        "distribute is not supported.")
 
-######################################################################
-# Install distribute
-ez = {}
+#######################################################################
+def install_pip():
+    print('')
+    print('Install pip')
+    print('')
+    try:
+        from urllib.request import urlopen
+    except ImportError:
+        from urllib2 import urlopen
 
-try:
-    from urllib.request import urlopen
-except ImportError:
-    from urllib2 import urlopen
+    tmp = tempfile.mkdtemp(prefix='buildout-dev-')
+    try:
+        get_pip = os.path.join(tmp, 'get-pip.py')
+        with open(get_pip, 'wb') as f:
+           f.write(urlopen('https://bootstrap.pypa.io/get-pip.py').read())
 
-# XXX use a more permanent ez_setup.py URL when available.
-exec(urlopen('https://bootstrap.pypa.io/ez_setup.py').read(), ez)
-ez['use_setuptools'](to_dir='eggs', download_delay=0)
+        if subprocess.call([sys.executable, get_pip]):
+            raise RuntimeError("pip failed.")
+    finally:
+        shutil.rmtree(tmp)
+    return_code = subprocess.call(
+        [sys.executable] + sys.argv + ['--no-clean']
+    )
+    sys.exit(return_code)
 
-import pkg_resources, setuptools
-setuptools_path = os.path.dirname(os.path.dirname(setuptools.__file__))
+try:
+    import pip
+except ImportError:
+    install_pip()
 
 ######################################################################
-# Install buildout
+print('')
+print('Latest dependencies')
+print('')
+if subprocess.call(
+    [sys.executable] +
+    ['-m', 'pip', 'install', '--upgrade', 'pip', 'setuptools', 'wheel'],
+    ):
+    raise RuntimeError("Latest dependencies failed.")
+######################################################################
+print('')
+print('Install buildout')
+print('')
 if subprocess.call(
     [sys.executable] +
     ['setup.py', '-q', 'develop', '-m', '-x', '-d', 'develop-eggs'],
-    env=dict(os.environ, PYTHONPATH=setuptools_path)):
+    ):
     raise RuntimeError("buildout build failed.")
 
+import pkg_resources
+
 pkg_resources.working_set.add_entry('src')
 
 import zc.buildout.easy_install
 zc.buildout.easy_install.scripts(
     ['zc.buildout'], pkg_resources.working_set , sys.executable, 'bin')
```

Comment:
Not part of your changes, but I can't notice a space in front of a comma and not mention it.